apiserver/params/network.go

```diff
@@ -1200,14 +1200,19 @@ type InterfaceAddress struct {
 type NetworkInfo struct {
 	// MACAddress is the network interface's hardware MAC address
 	// (e.g. "aa:bb:cc:dd:ee:ff").
-	MACAddress string `json:"mac-address"`
+	MACAddress string `json:"mac-address" yaml:"mac-address"`
 
 	// InterfaceName is the raw OS-specific network device name (e.g.
 	// "eth1", even for a VLAN eth1.42 virtual interface).
-	InterfaceName string `json:"interface-name"`
+	InterfaceName string `json:"interface-name" yaml:"interface-name"`
 
 	// Addresses contains a list of addresses configured on the interface.
-	Addresses []InterfaceAddress `json:"addresses"`
+	Addresses []InterfaceAddress `json:"addresses" yaml:"addresses"`
+
+	// These fields are for preserving backward compatibility with agents that
+	// anticipate the old YAML format. Remove for Juju 3/4.
+	MACAddressX    string `json:"macaddress" yaml:"macaddress"`
```

Comment:
I'd rather not see these on the apiserver/params struct, vs just the client serialization struct.

At the very least, I'd also prefer this to be:
`
```suggestion
	MACAddressX    string `json:"-" yaml:"macaddress"`
```

That way we don't introduce fields into the JSON format that we don't want to support.

Reply:
What happens if you do it this way is that you loose the fields over the API server->client trip and they are always empty for any format.

The correct solution is to have a display type that the payload is converted to. I'll close this.